### ISSUE TITLE
Repository size exceeds 50 MB breaking jsDelivr

### ISSUE THREAD

**Issue by jacekkow** [openlayers/openlayers.github.io#113]
Today I got hit by surprise that URL for version 5.3.0 of ol.js mentioned in docs some time ago stopped working.

https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/v5.3.0/build/ol.js
returns:
```
Package size exceeded the configured limit of 50 MB. Try https://github.com/openlayers/openlayers.github.io/tree/master/en/v5.3.0/build/ol.js instead.
```

I expected these URLs to last, but apparently due to lack of versioning, "master" branch of openlayers.github.io exceeded 50 MB limit imposed by jsDelivr, causing some files to be missing from CDN.

Please either contact jsDelivr to increase this limit or add some kind of tagging scheme where individual tag contains only a single version of library, staying within the limit.

**Comment by MartinKolarik** [comment]
Hi, Martin from jsDelivr here.

jsDelivr is normally able to fetch individual files instead of the whole branch, so this limit doesn't usually apply unless you request a directory listing. What most likely happened here is that there was a temporary issue with fetching individual files, which triggered fetching the whole tarball as a fallback, which then failed with this error. The link should now work again.

**Comment by RadekHavelka** [comment]
please fix it, the whole CDN is broken now

**Comment by symbioquine** [comment]
@MartinKolarik It's awesome that jsDelivr is directly engaged on this issue!

It looks like https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/v6.5.0/build/ol.js is failing consistently now with the error described above.

```sh
$ curl -i "https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/v6.5.0/build/ol.js"
HTTP/2 403 
date: Thu, 29 Apr 2021 17:26:24 GMT
content-type: text/plain; charset=utf-8
content-length: 150
access-control-allow-origin: *
access-control-expose-headers: *
timing-allow-origin: *
cache-control: public, max-age=604800, s-maxage=43200
cross-origin-resource-policy: cross-origin
x-content-type-options: nosniff
strict-transport-security: max-age=31536000; includeSubDomains; preload
etag: W/"96-wdVoMqXYseBtQD1/l2RYk5ymBLo"
age: 16859
x-served-by: cache-fra19151-FRA, cache-sea4439-SEA
x-cache: HIT, HIT
vary: Accept-Encoding
cf-cache-status: HIT
cf-request-id: 09c043f556000002d45f812000000001
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 647a3c355b1102d4-SEA
alt-svc: h3-27=":443"; ma=86400, h3-28=":443"; ma=86400, h3-29=":443"; ma=86400

Package size exceeded the configured limit of 50 MB. Try https://github.com/openlayers/openlayers.github.io/tree/master/en/v6.5.0/build/ol.js instead.
```

Oddly, it was working for this version a few days ago... (and maybe as recently as yesterday)

**Comment by MartinKolarik** [comment]
@symbioquine please try now and if there are any other still broken, let me know.

**Comment by symbioquine** [comment]
Yay, looks like that version is working now. Thanks @MartinKolarik!

There's also a few other versions which are broken. See those with HTTP status code 403 below;

```sh
$ VERSIONS=`curl --silent "https://github.com/openlayers/openlayers.github.io/tree/master/en" | grep -Po '/openlayers/openlayers.github.io/tree/master/en/\K[^"]+'`
$ for v in $VERSIONS; do echo $v - $(curl --silent -I "https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/$v/build/ol.js" | head -n 1); done
latest - HTTP/2 403 
main - HTTP/2 200 
master - HTTP/2 200 
v3.20.1 - HTTP/2 200 
v4.6.5 - HTTP/2 403 
v5.0.0 - HTTP/2 200 
v5.0.2 - HTTP/2 200 
v5.0.3 - HTTP/2 200 
v5.1.2 - HTTP/2 200 
v5.1.3 - HTTP/2 200 
v5.2.0 - HTTP/2 403 
v5.3.0 - HTTP/2 200 
v6.0.0 - HTTP/2 200 
v6.0.1 - HTTP/2 200 
v6.1.0 - HTTP/2 200 
v6.1.1 - HTTP/2 200 
v6.2.0 - HTTP/2 200 
v6.2.1 - HTTP/2 200 
v6.3.0 - HTTP/2 200 
v6.3.1 - HTTP/2 403 
v6.4.0 - HTTP/2 200 
v6.4.1 - HTTP/2 200 
v6.4.2 - HTTP/2 200 
v6.4.3 - HTTP/2 200 
v6.5.0 - HTTP/2 200
```

It would also probably be a good idea to figure out a strategy to ensure that future releases don't have this issue...

**Comment by MartinKolarik** [comment]
Thanks, they should all work now and we'll push a fix for the underlying issue shortly. Still, it would be best if openlayers switched to using properly tagged releases in the future, as the current approach of having everything in one branch has some performance and reliability limits.

**Comment by symbioquine** [comment]
Agreed, and thanks again @MartinKolarik!

Seems like this issue should stay open to track that - or at least until someone more familiar with the rational for the current setup can comment...

**Comment by ahocevar** [comment]
Thanks for the help, @MartinKolarik.

The reason for the current setup is that jsdelivr points to our website repository (openlayers.github.io), and takes the legacy builds from there. We do not store the legacy build in release tags of the main repository (openlayers) and the npm package (ol), because due to the nature of OpenLayers, carrying around all the code in a full build does not make sense and we discourage its use.

**Comment by ahocevar** [comment]
@MartinKolarik This issue has resurfaced, see https://github.com/openlayers/openlayers/issues/13606#issuecomment-1110739638. Any chance you could help us again to fix this?

**Comment by MartinKolarik** [comment]
@ahocevar as I understand this was about missing files. It'll work automatically once you add them in a new version (we don't support changes in existing releases).

**Comment by ahocevar** [comment]
Thans @MartinKolarikk, I'm going to publish a new release then.

**Comment by jpmens** [comment]
Today I got hit by surprise that URL for version 6.1.1 used by a bit older version of Traccar no longer exists. I have the feeling this issue matches what I am seeing. 

A Firefox browser reports the following in the console:

![rabbit-9483](https://user-images.githubusercontent.com/60706/186128607-58a78943-cca7-4778-be76-6a1aaadc5d7e.png)

whereas Safari reports

![rabbit-9487](https://user-images.githubusercontent.com/60706/186128676-69844c67-01e3-46f1-8523-1925fb6836b3.png)

An attempt to access the URL `https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/v6.1.1/css/ol.css` leads to the diagnostic

```
Package size exceeded the configured limit of 50 MB. Try https://github.com/openlayers/openlayers.github.io/tree/master/en/v6.1.1/build/ol.js instead.
```

which is how I found my way here.

The directory `en/v6.1.1` doesn't exist.

Is there any way we could have that reinstated, please? I don't see how we can easily change the client application.

Thank you.

**Comment by jldelacalle** [comment]
The same problems occur with the 6.3.1 version.

**Comment by ricef** [comment]
Same problem version 6.12!!!!
How it's possible delete whole repository!!!!

**Comment by ahocevar** [comment]
For clarification: this is a different issue. We deleted these resources when we updated the website for the v7.0.0 release. We only left the last v6 release there, v6.15.1. We did the same when we released v6.0.0 with previous versions. Remember: these urls were only provided for quick prototyping, and wherever we showed it we had a big fat warning to not use these in production.
